### DISCUSSION TITLE
How many possible filters are even available?

### DISCUSSION THREAD

**Discussion by shakhrillodev** [aiogram/aiogram#1375] [Q&A] [answered]
I am using aiogram 2.4. 
I found out that there are filters like:
@dp.message_handler(text="Hi")
@dp.message_handler(CommandStart())
But I could not find any info about all bot filter types(e.g: text="Hi", user_id=1231232414)
Please help.
Really appreciated.

**Comment by JrooTJunior** [comment, marked as answer]
Infinity because you can implement your own filter.
All builtin filters implemented at this module (v2.25.2): https://github.com/aiogram/aiogram/blob/dev-2.x/aiogram/dispatcher/filters/builtin.py

Also i recommend you to update to aiogram>=3.0 (3.2 is available at this time)

**Reply by shakhrillodev** [reply]
Thanks. I really appreciate your help. Also have you ever thought of making a tutorial online like on youtube. Like they "No one uses a tool better than its creator". It would have been really helpful.
